### ISSUE TITLE
Differences between usage with python2 and python3

### ISSUE THREAD

**Issue by JochemGit** [dsuch/pymqi#207]
Hello @SeyfSV ,

I implemented a lot of functions with python3, but now with python2 some stuff does not seem to be implemented. Here is a small script as an example:

```
import pymqi

queue_manager = 'QM4'

# MQCMD_INQUIRE_LISTENER
args_dis_l = {pymqi.CMQCFC.MQCACH_LISTENER_NAME: '*',
              pymqi.CMQCFC.MQIACF_LISTENER_ATTRS: pymqi.CMQCFC.MQIACF_ALL}


qmgr = pymqi.connect(queue_manager)

pcf = pymqi.PCFExecute(qmgr)

print(args_dis_l)
res = pcf.MQCMD_INQUIRE_LISTENER(args_dis_l)
print(res)

qmgr.disconnect()
```

With python3 the output is
```
{3554: '*', 1222: 1009}
[{3554: b'SYSTEM.DEFAULT.LISTENER.TCP                     ', 1601: 2, 1501: 2, 1522: 0, 3552: b'                                                ', 1602: 0, 3555: b'                                                                ', 2027: b'2020-06-16  ', 2028: b'15.27.10'}]
```

With python2
```
{3554: '*', 1222: 1009}
Traceback (most recent call last):
  File "./foo.py", line 17, in <module>
    res = pcf.MQCMD_INQUIRE_LISTENER(args_dis_l)
  File "build/bdist.linux-x86_64/egg/pymqi/__init__.py", line 2684, in __call__
  File "build/bdist.linux-x86_64/egg/pymqi/__init__.py", line 2865, in unpack
NotImplementedError: Unpack for type ((4,)) not implemented
```
Versions are Python 2.7.17, Python 3.6.9 and pymqi is not the last version commited half a day ago, but the version before.
Please have a look. Is this a problem of pymqi or of the environment?

**Comment by SeyfSV** [comment]
It's a problem of pymqi.

**Comment by SeyfSV** [comment]
This one can be fixed: pymqi should unpack parameter's type before comparing.
But should we support python 2.7, that reached the end of its life on January 1st, 2020?

**Comment by dsuch** [comment]
The core Python team stopped to support Python 2.7, this is true, but this does not change much for PyMQI, which is mostly used by larger companies that are usually at least 5-10 years behind the open-source world. Consider RHEL 7 at the very least.

There will be thousands and millions of Python 2.7 installations for years to come. Hence, yes, we continue to support Python 2.7.

**Comment by SeyfSV** [comment]
This [charts](https://dev.to/hugovk/python-version-share-over-time-6-1jb8) show, that python 2.7 is still widely used.

**Comment by dsuch** [comment]
Yes and I would expect for it to be much more significant among the users of PyMQI - you can see yourself how many people here do not even use PyPI, usually for some "security reasons" fantasised about by their IT.

**Comment by SeyfSV** [comment]
OK!
Please, take a look on this [gh159-actions-unit-testing](https://github.com/dsuch/pymqi/tree/gh159-actions-unit-testing) branch and #159 issue. It is really needed for us to simplify cross version testing.

**Comment by dsuch** [comment]
That ticket, and the test framework, look impressive indeed! I am all for including it in the core.

I am in a peculiar situation where you and @JoshMcMQ do much more than me in PyMQI.

I had been maintaining it for around 15 years but now my area of focus is [Zato](https://zato.io) and I tend to work with PyMQI only as it pertains to our ongoing Zato-based projects.

Also, from to time, I tend to remind people to be on their best behaviour because I truly dislike the fact that some people here do not exhibit good manners when they first open their tickets.

Other than that, I am currently simply trying not to get into your or JoshMcMQ's way.

**Comment by JoshMcMQ** [comment]
@SeyfSV is the real star here with all the recent enhancements.   I'm just a MQ Admin lending a hand answering questions and helping were I can.

**Comment by SeyfSV** [comment]
@JochemGit , I fixed this and pushed it to master. But I found other issue with python 2.7 and raw PCF :(. Order of parameters has restrictions, and required parameters, that we put in top of dictionary, can move to any other place.
`dictionary` processing was [changed](https://portingguide.readthedocs.io/en/latest/dicts.html) some time ago and I used this behaviour:
> Additionally, CPython 3.6+ uses a new implementation of dictionaries, which makes them appear sorted by insertion order (though you can only rely on this behavior in Python 3.7+).

I need some time to come up with fix.

**Comment by JochemGit** [comment]
Thank you for your quick work, @SeyfSV !
I will test it and keep you informed.

**Comment by SeyfSV** [comment]
@AlexandreYang @salapat11 , do you use python 2.7? Do you have issues?

**Comment by salapat11** [comment]
We use python 3.8.

**Comment by dsuch** [comment]
@SeyfSV - Python 2.7 and newer have collections.OrderedDict, maybe this will be helpful?

**Comment by AlexandreYang** [comment]
@SeyfSV I used python 3.8 to test https://github.com/dsuch/pymqi/issues/204 . Trying now with 2.7.

**Comment by SeyfSV** [comment]
@dsuch yes, it can help (I think), but this is not backward compatibility.

**Comment by AlexandreYang** [comment]
It's working now with your latest commit.

```
nosetests tests/test_pcf.py:TestPCF.test_mqcfgr_mqcfin64_mqcfil64  tests/test_pcf.py:TestPCF.test_unpack_group
..
----------------------------------------------------------------------
Ran 2 tests in 0.107s

OK
```

But without your latest commit, there are those errors (as expected):
- `ImportError: No module named typing`
- `NotImplementedError: Unpack for type ((4,)) not implemented`

**Comment by AlexandreYang** [comment]
> Please, take a look on this [gh159-actions-unit-testing](https://github.com/dsuch/pymqi/tree/gh159-actions-unit-testing) branch and #159 issue. It is really needed for us to simplify cross version testing.

Agree that automated CI testing is needed to avoid regression and cross version / matrix testing.

Reviving this PR would be great https://github.com/dsuch/pymqi/pull/163

**Comment by JochemGit** [comment]
Hi @SeyfSV ,

the fix works very well for me. I changed my modules all to the new "style" with `attrs = [] ...` and now it works for me for all objects I need with python2 and python3.
I do not know, if it is important for the fix, the objects I use are authority records, channels, chlauth records, listeners, processes, qmgrs, queues and services. Additionally I use refresh security.

Thank you very much!

**Comment by SeyfSV** [comment]
Fixed, but #211
